tpv/rules/gateway.py

```diff
@@ -43,10 +49,20 @@ def reload_destination_mapper(path=None):
     return mapper
 
 
-def map_tool_to_destination(app, job, tool, user, tpv_config_files, job_wrapper=None, resource_params=None,
-                            workflow_invocation_uuid=None):
-    global ACTIVE_DESTINATION_MAPPER
-    if not ACTIVE_DESTINATION_MAPPER:
-        ACTIVE_DESTINATION_MAPPER = setup_destination_mapper(app, tpv_config_files)
-    return ACTIVE_DESTINATION_MAPPER.map_to_destination(app, tool, user, job, job_wrapper, resource_params,
+def map_tool_to_destination(
+    app,
+    job,
+    tool,
+    user,
+    tpv_config_files: Union[List[str], str],
+    job_wrapper=None,
+    resource_params=None,
+    workflow_invocation_uuid=None,
```

Comment:
```suggestion
    workflow_invocation_uuid=None,
    referrer=None,
```

Reply:
@nuwang 

- Can this really be `None`? then we can not use it as key in the dict
- Currently the reload function can use the config files from the keys of `ACTIVE_DESTINATION_MAPPERS`. If we change to referrer we probably need another dict? Ideas?

Reply:
@bernt-matthias I think you're right. For now, I've defaulted the referrer value to "tpv_dispatcher" just so we don't have to pass the referrer parameter in all tests, but I think we can safely assume that it's never null.

I've added a fix+test. Let me know whether this looks ok to you.